app/models/miq_worker/runner.rb

```diff
@@ -130,6 +133,11 @@ def start
     end
   end
 
+  def recover_from_provider_unreachable
+    sleep 2.minutes
+    do_before_work_loop
```

Comment:
Is this necessary? As far as I can see, we previously promised to call `do_before_work_loop` exactly once... and `do_work` will itself recover the failed `@tid` (but anyway.. that's a subclass concern).